### ISSUE TITLE
TypeError: e.getSetCookie is not a function

### ISSUE THREAD

**Issue by grzegorzpokorski** [vercel/next.js#56947]
### Link to the code that reproduces this issue

https://github.com/grzegorzpokorski/nextjs-getsetcookie-error

### To Reproduce

1. clone repo
2. `pnpm i`
3. `pnpm dev`
4.  open `localhost:3000` in browser
5. should occur error

### Current vs. Expected behavior

error should not occur

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP Fri Jan 27 02:56:13 UTC 2023
Binaries:
  Node: 18.11.0
  npm: 8.19.2
  Yarn: 1.22.19
  pnpm: 8.9.2
Relevant Packages:
  next: 13.5.6-canary.2
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

**Comment by ze-kel** [comment]
Just wanted to open this. Breaked lucia-auth in my project.

I also digged a bit more and discovered that this issue is not preset when using Node v16, while being present on Node Stable(v19.3.0).

[Codesandbox with it working on Node 16.](https://codesandbox.io/p/sandbox/determined-mcnulty-6m2lq4?file=%2Fnext.config.js%3A1%2C1)

**Comment by zfreiter** [comment]
I think I have a similar issue -> TypeError: responseHeaders.getSetCookie is not a function.
This is my first time created a Nextjs app today. I have Node v18.18.2 and had the most current NextJS version. When I got to NextJS version 13.5.4 the error goes away.

**Comment by grzegorzpokorski** [comment]
@huozhi  you have right! Upgrading node to 18.16.1 resolved issue. App also successfully builds on Vercel ❤

**Comment by jawkhan** [comment]
Upgrading to the latest version of node worked for me. I had version 19.1.0 and upgraded to version 21.0.0.

I used node version manager (nvm) and ran the following command:
`nvm install node`

You can follow these instructions to install nvm:
https://www.freecodecamp.org/news/node-version-manager-nvm-install-guide/

**Comment by abdolrhman** [comment]
upgrade fix the issue, use latest node --v v18.18.2

**Comment by MuhammadAli-123** [comment]
Yes, After Upgrading the Node Version to Latest will solve this problem.

**Comment by huozhi** [comment]
We patched a fix https://github.com/vercel/next.js/releases/tag/v13.5.6, please upgrade to the new version

**Comment by goudete** [comment]
> I can repro this with node 18.11, you can upgrade your nodejs version 18.16 it will fix the problem

This worked for me. THANK YOU!

**Comment by Matdweb** [comment]
It also worked for me! And I am running `node 20.9.0`!
But It would be great to put this in the documentation because it took me a while to get to this Issue and I had no idea 😢

**Comment by Matdweb** [comment]
> It also worked for me! And I am running `node 20.9.0`! But It would be great to put this in the documentation because it took me a while to get to this Issue and I had no idea 😢

I actually encountered issues with API routes in `Next.js 13`, and it was quite challenging because they didn't work at all, and I had no idea why.

**Comment by huozhi** [comment]
If you're running 13.5.6, the allowed engine versions of node.js should be fine

https://github.com/vercel/next.js/issues/56947#issuecomment-1769572453
